apps/angular/21-anchor-navigation/src/app/nav-button.component.ts

```diff
@@ -1,17 +1,23 @@
 /* eslint-disable @angular-eslint/component-selector */
-import { Component, Input } from '@angular/core';
+import { Component, input } from '@angular/core';
+import { RouterLink } from '@angular/router';
+
 @Component({
   selector: 'nav-button',
   standalone: true,
+  imports: [RouterLink],
   template: `
-    <a [href]="href">
-      <ng-content></ng-content>
+    <a
+      [routerLink]="route()"
+      [fragment]="fragment() === '' ? undefined : fragment()">
+      <ng-content />
     </a>
```

Comment:
What you can do as well is to have a selector like a[nav-button], this way the routerLink is on the parent, and this component is just used for design purposes, since we don't add any logic to the route.